boofuzz/socket_connection.py

```diff
@@ -93,8 +93,10 @@ def __init__(self,
         self.l2_dst = l2_dst
         self._udp_broadcast = udp_broadcast
         self.server = server
-        self.keyfile = keyfile
-        self.certfile = certfile
+        self.sslcontext = sslcontext
+        self.server_hostname = server_hostname
+        assert(not self.server or self.sslcontext, "parameter sslcontext is required when server=True.")
+        assert(self.sslcontext or self.server, "You did neither specify a sslcontext nor server_hostname. Can't proceed without making you vulnerable to MITM attacks. If this is what you want, then specify a sslcontext.")
```

Comment:
1. Looks like these checks should only apply when the protocol is `ssl`
2. Use `raise ValueError...` to match current approach. See example a few lines below in the code: `            raise ValueError("__init__() argument port required for protocol {0}".format(self.proto))`
3. Be explicit about what is being checked. E.g. `self.server is not None and self.sslcontext is None`
4. The error explanations are a bit confusing. Try rewording them for clarity. It is helpful to use the same format for each so that the semantic difference is obvious. You might also match the ValueError example I cited. E.g., `__init__() argument sslcontext is required when server == True`.
5.  Should the second assert `assert(self.sslcontext or self.server, ...` be checking `self.server_name` instead of `self.server`?

Reply:
1. yes
2. ValueError is of course the better way, I currently need to fuzz a TLS server/client and did some "hacking"... changed it, see below
3. done (flake8 does report this, too)
4. I'm not native english speaker, hopefully it's clearer now!
5. correct, that was a mistake/bug